File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> charStack = new Stack<>();
        
        int i=0;
        while (i<s.length()) {
            char c = s.charAt(i);
            
            if (c == '(')
                charStack.push(')');
            if (c == '[')
                charStack.push(']');
            if (c == '{')
                charStack.push('}');
            
            if (c == ')' || c == ']' || c == '}') {
                if (charStack.empty()) {
                    return false;
                }
                
                if (!(charStack.pop() == c)) {
                    return false;
                }
            }
            i++;
        }
        return charStack.empty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        openBracket = ('(', '{', '[')
        stack = []
        
        for c in s:
            if c in openBracket:
                if c == '(':
                    stack.append(')')
                if c == '{':
                    stack.append('}')
                if c == '[':
                    stack.append(']')
            else:
                
                if not stack or c != stack.pop():
                    return False
        
        return not stack
                
        